### PR TITLE
HDDS-9037. Provide documentation for Decommissioning in Ozone

### PR DESCRIPTION
**Pull request by aryangupta1998**
## What changes were proposed in this pull request?

This PR provides the documentation for decommissioning of OM, SCM and Datanode.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9037

## How was this patch tested?

Tested Manually.

### CODE REVIEW

File: hadoop-hdds/docs/content/feature/Decommission.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+#DataNode Decommission
+
+The DataNode decommission is the process that removes the existing DataNode from the Ozone cluster while ensuring that the new data should not be written to the decommissioned DataNode. When you initiate the process of decommissioning a DataNode, Ozone automatically ensures that all the storage containers on that DataNode have an additional copy created on another DataNode before the decommission completes.
+
```
[the remark this answers is not in the document]
Comment (aryangupta1998) [reply hadoop-hdds/docs/content/feature/Decommission.md:45]: I have added a few more lines that describe the process of decommissioning briefly.

File: hadoop-hdds/docs/content/feature/Decommission.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ozone admin datanode recommission [-hV] [-id=<scmServiceId>]
+       [--scm=<scm>] [<hosts>...]
+```
+
+## OM Decommission
+
+Ozone Manager (OM) decommissioning is the process in which you gracefully remove one of the OM from the OM HA Ring.
+
+To decommission an OM and remove the node from the OM HA ring, the following steps need to be executed.
+1. Stop the OzoneManager process only on the node which needs to be decommissioned. <p> **Note -** Do not stop the decommissioning OM if there are
```
Comment (ChenSammi) [inline hadoop-hdds/docs/content/feature/Decommission.md:65]: Is stop the OM a must step before decommission?

**Comment by ChenSammi** [comment]
Hi @aryangupta1998 , could you help to create a follow up JIRA for Chinese version document of this decommission?

File: hadoop-hdds/docs/content/feature/Decommission.md (hunk 2 above)
Comment (aryangupta1998) [reply hadoop-hdds/docs/content/feature/Decommission.md:65]: Yes, if we even won't stop, after adding the property `ozone.om.decommissioned.nodes.<omServiceId>` mentioned in step 2, the OM node will be automatically stopped.

File: hadoop-hdds/docs/content/feature/Decommission.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+Storage Container Manager (SCM) decommissioning is the process in which you can gracefully remove one of the SCM from the SCM HA Ring.
+
+To decommission a SCM and remove the node from the SCM HA ring, the following steps need to be executed.
+```shell
+ozone admin scm decommission [-hV] [-id=<scmServiceId>] -nodeid=<nodeId>
+                                    [--scm=<scm>]
+```
+You can obtain the 'nodeid' by executing this command, **"ozone admin scm roles"**
+
+**Note -** If you want to decommission a **primordial** scm, first change the primordial scm.
```
[the remark this answers is not in the document]
Comment (aryangupta1998) [reply hadoop-hdds/docs/content/feature/Decommission.md:87]: We can't directly decommission a primordial scm so we make some other scm host as primordial scm node by updating the property `ozone.scm.primordial.node.id` and then decommission the original primordial scm. Same mentioned [here](https://docs-stage.cloudera.com/cdp-private-cloud-base/7.1.9/ozone-storing-data/topics/ozone-decommission-primordial-scm.html).

**Comment by aryangupta1998** [comment]
> Hi @aryangupta1998 , could you help to create a follow up JIRA for Chinese version document of this decommission?

Sure.

File: hadoop-hdds/docs/content/feature/Decommission.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChenSammi) [reply hadoop-hdds/docs/content/feature/Decommission.md:87]: Is this the limitation posted by CM?  In the community version, I think we don't have such limitation. You can refer to scm-decommission.robot test.

File: hadoop-hdds/docs/content/feature/Decommission.md (hunk 2 above)
Comment (ChenSammi) [reply hadoop-hdds/docs/content/feature/Decommission.md:65]: If OM stop is a must step. Then it's kind conflict with the Note that "Do not stop the decommissioning OM if there are only two OMs in the ring as both the OMs would be needed to reach consensus to update the Ratis configuration". 
If change the property ozone.om.decommissioned.nodes.<omServiceId> will cause OM to automatically stop, then it looks like it's not feasible to decommission one OM when there are only two OM instances left.  
I'm a little confused at the point. Could you do a further explain?

File: hadoop-hdds/docs/content/feature/Decommission.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChenSammi) [reply hadoop-hdds/docs/content/feature/Decommission.md:87]: Hi @nandakumar131 , can you take a look of this statement?

**Review by sodonnel — APPROVED** [review]
LGTM from the Datanode Decommissioning perspective. Once you reach a conclusion on the other points we can commit.

File: hadoop-hdds/docs/content/feature/Decommission.md (hunk 2 above)
Comment (ChenSammi) [reply hadoop-hdds/docs/content/feature/Decommission.md:65]: @aryangupta1998 , please remove this step 1, as it's not a must step for upstream Ozone.

Comment (aryangupta1998) [reply hadoop-hdds/docs/content/feature/Decommission.md]: Removed Step 1.

**Pull request merged**